### PR TITLE
Register GitHub Authentication provider for the vscode GitHub PR plugin

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
* Register authentication provider via the [authentication plugin API](https://github.com/eclipse-theia/theia/pull/8402) based on the Che GitHub oAuth, so the native command `Github sign in` from the PR plugin calls the registered provider.
* Remove the temporary command that injects the GitHub token to the vscode PR plugin settings and ask to reload the browser page.
* Update the GitHub Plugin API with `getUser()` function which returns GitHub user info.

depends on https://github.com/eclipse/che-plugin-registry/pull/640
### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/17684
depends on https://github.com/eclipse-theia/theia/pull/8402

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

**Comment by azatsarynnyy** [comment]
depends on https://github.com/eclipse/che/issues/17550

**Comment by vinokurig** [comment]
depends on https://github.com/eclipse/che/issues/16741

**Comment by vinokurig** [comment]
[crw-ci-test]

File: plugins/github-auth-plugin/src/github-auth-plugin.ts [hunk 1]
```diff
@@ -11,22 +11,54 @@
...
+                        return [session];
+                    } else {
+                        return [];
+                    }
+                },
+                login: async (scopes: string[]) => {
+                    const githubUser = await che.github.getUser();
+                    session = {
+                        id: 'github-session',
+                        accessToken: await che.github.getToken(),
```
Comment (benoitf) [inline plugins/github-auth-plugin/src/github-auth-plugin.ts:34]: what happen on fresh install, I'm not sure che.github.getToken() or che.github.getUser() is returning something ?

Comment (vinokurig) [reply plugins/github-auth-plugin/src/github-auth-plugin.ts:34]: In this case the GitHub authentication page appears:
https://github.com/eclipse/che-theia/blob/2f11124e7de0e50feed25fddf91a9058c37a2a1b/extensions/eclipse-che-theia-plugin-ext/src/browser/oauth-utils.ts#L101-L117
Then if user authenticates it the GitHub token is setup and these functions work as expected.

**Comment by vinokurig** [comment]
[crw-ci-test]

File: generator/tests/init-sources/assembly-example/configs/root-compilation.tsconfig.json [hunk 2]
```diff
@@ -164,4 +164,4 @@
       "path": "../examples/assembly/compile.tsconfig.json"
     }
   ]
-}
+}
```
Comment (azatsarynnyy) [inline generator/tests/init-sources/assembly-example/configs/root-compilation.tsconfig.json:167]: please, revert it to have the checks passed

image

**Review by azatsarynnyy — APPROVED** [review]

File: generator/tests/init-sources/assembly-example/configs/root-compilation.tsconfig.json (hunk 2 above)
Comment (vinokurig) [reply generator/tests/init-sources/assembly-example/configs/root-compilation.tsconfig.json]: Done, thank you!

**Comment by vinokurig** [comment]
1. Register authentication provider via the authentication plugin API based on the Che GitHub oAuth, so the native command Github sign in from the PR plugin calls the registered provider.
2. Remove the temporary command that injects the GitHub token to the vscode PR plugin settings and ask to reload the browser page.
3. Update the GitHub Plugin API with getUser() function which returns GitHub user info.

**Pull request merged**